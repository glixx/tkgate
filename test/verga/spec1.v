module top;
  integer i;
  reg a,b;
  wire x;

  initial
    $monitor("%4d: a=%b b=%b - x=%b",$time,a,b,x);

  initial
    for (i = 0;i < 4;i = i + 1)
      begin
	{a,b} = i;
	#50;
//	$finish();
      end

  foo f(a,b,x);

endmodule

module foo(a,b,x);
input a,b;
output x;

  specify
    if (a) (a *> x) = 10;
    if (!a) (a *> x) = 21;
    (b *> x) = 12;
  endspecify

  assign x = a ^ b;

endmodule
