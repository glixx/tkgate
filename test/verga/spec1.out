   0: a=0 b=0 - x=x
  21: a=0 b=0 - x=0
  50: a=0 b=1 - x=0
  62: a=0 b=1 - x=1
 100: a=1 b=0 - x=1
 110: a=1 b=0 - x=0
 112: a=1 b=0 - x=1
 150: a=1 b=1 - x=1
 162: a=1 b=1 - x=0
